cloudferry_devlab/cloudferry_devlab/tests/base.py

```diff
@@ -290,31 +304,46 @@ def get_sec_group_id_by_tenant_id(self, tenant_id):
         return [i['id'] for i in sec_group_list['security_groups']
                 if i['tenant_id'] == tenant_id]
 
-    def get_users_keypairs(self):
+    def get_users_attributes(self, attr):
```

Comment:
This should be two separate methods: for keypairs and volumes. Please refactor and reuse as much as possible.